### PR TITLE
initial

### PR DESCRIPTION
**Pull request by mtsyhanok**

### CODE REVIEW

File: src/main/resources/story/shop_api/GetCategories.story [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Story: GET /categories/ test
+
+Scenario: check response body
+When I issue a HTTP GET request for a resource with the URL 'https://api.predic8.de/shop/categories/'
+Then the response code is equal to '200'
+When I save response body to the scenario variable 'body'
+Then a JSON element from '${body}' by the JSON path '$' is equal to '{
+"categories": [
```
Comment (mtsyhanok) [inline src/main/resources/story/shop_api/GetCategories.story]: Can I read somehow expected json from file? It's not convenient to store it inside story as it's bulky

File: src/main/resources/story/shop_api/GetCategoriesID.story [hunk 2]
```diff
@@ -0,0 +1,77 @@
+Story: GET /categories/{id} test
+
+Scenario: check response body
+When I issue a HTTP GET request for a resource with the URL 'https://api.predic8.de/shop/categories/<category_id>'
+Then the response code is equal to '200'
+When I save response body to the scenario variable 'body'
+Then a JSON element from '${body}' by the JSON path '$' is equal to <expected response>
+Examples:
```
Comment (mtsyhanok) [inline src/main/resources/story/shop_api/GetCategoriesID.story:8]: How can I implement DDT approach here? I'd like to perform GET request for several <category_id> and for every category check that actual json matches expected.
It looks like I can't store json inside 'Examples'

File: src/main/resources/story/shop_api/PostCustomers.story [hunk 3]
```diff
@@ -0,0 +1,16 @@
+Story: POST /cusomers/ test
+
+Scenario: check response body
+When I send HTTP POST to the relative URL '/customers/' with content: '{
```
Comment (mtsyhanok) [inline src/main/resources/story/shop_api/PostCustomers.story]: Is there step for sending HTTP POST to the absolute URL?
I added property `web-application.main-page-url=https://api.predic8.de:443/shop` to environment.properties but this step still fails

File: src/main/resources/story/shop_api/PostCustomers.story [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  "firstname": "#{generate(Name.firstName)}",
+  "lastname": "#{generate(Name.lastName)}"
+}'
+Then the response code is equal to '201'
+When I save response body to the scenario variable 'body'
+Then a JSON element from '${body}' by the JSON path '$' is equal to '{
+  "firstname": "Freddy",
+  "lastname": "Meyers",
+  "customer_url": "/shop/customers/374"
```
Comment (mtsyhanok) [inline src/main/resources/story/shop_api/PostCustomers.story]: Can I somehow insert regexp into the line `customer_url": "/shop/customers/374` of expected json to check that actual json matches  `customer_url": "/shop/customers/\d+`

File: src/main/resources/story/shop_api/GetCategories.story (hunk 1 above)
Comment (mtsyhanok) [reply src/main/resources/story/shop_api/GetCategories.story]: How can I check whether actual json matches expected json schema?

File: src/main/resources/story/shop_api/PostCustomers.story (hunk 3 above)
Comment (AngelinaMarchyk) [reply src/main/resources/story/shop_api/PostCustomers.story]: 1. Example of absolute url usage: When I issue a HTTP HEAD request for a resource with the URL 'https://google.com/'
2. To use steps with parameter 'relative url' you need to open main-page-url. As example:
```
Given I am on the main application page

```

File: src/main/resources/story/shop_api/GetCategoriesID.story (hunk 2 above)
Comment (AngelinaMarchyk) [reply src/main/resources/story/shop_api/GetCategoriesID.story:8]: https://git.epam.com/valery_yatsynovich/vividus/wikis/Handbook/Expressions/Resources 
in theory you can hide json in txt file, but usually we don't check the whole json, just some elements and they can be reached by json path

Comment (apaulenkova) [reply src/main/resources/story/shop_api/GetCategoriesID.story:8]: You can also use this step 'Then a JSON element by the JSON path '$jsonPath' is equal to '$expectedData'$options'  and here you can skip step6 as this step will refer to response json from the previous 'When I send...' step

File: src/main/resources/story/shop_api/PostCustomers.story (hunk 4 above)
Comment (apaulenkova) [reply src/main/resources/story/shop_api/PostCustomers.story]: You can use step 'When I save a JSON element from 'json' by JSON path 'jsonPath' to scopes variable 'variableName'' to save the value from line#13, then use any of those steps to validate that it matches:
Then the response body $comparisonRule '$content' (insert regex in $content)
Then a JSON element by the JSON path '$jsonPath' is equal to '$expectedData' (insert regex in $expectedData)

File: src/main/resources/story/shop_api/PostCustomers.story (hunk 3 above)
Comment (mtsyhanok) [reply src/main/resources/story/shop_api/PostCustomers.story]: property `web-application.main-page-url` refers to web UI actions.
for API I set up `rest-api.http.endpoint=https://api.predic8.de/shop/`

**Review by apaulenkova — APPROVED** [review]

**Pull request merged**